### ISSUE TITLE
Using json database type to store data

### ISSUE THREAD

**Issue by JindrichPilar** [dbojdo/Doctrine-JMS-JSON#2]
I propose to change the format of stored data to true json, which will allow using json functions inside database. 

I propose to change:
```php
return sprintf('%s::%s', $type, $this->serializer()->serialize($value, 'json'));
```
To:
```php
return sprintf('{"class": "%s", "data": %s', $type, $this->serializer()->serialize($value, 'json')); 
```
 Note no ``"`` around second ``%s``

And change 
```php
    public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform)
    {
        return $platform->getClobTypeDeclarationSQL($fieldDeclaration);
    }
```
To 
```php
    public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform)
    {
        return $platform->getJsonTypeDeclarationSQL($fieldDeclaration);
    }
```

This changes were not tested, they are just a suggestion.

**Comment by dbojdo** [comment]
@JindrichPilar That's a great idea. I was thinking about this already. However, I think we have to introduce a new type, rather than change the existing one, as this is a breaking change.

**Comment by dbojdo** [comment]
@JindrichPilar It's been delivered in v2.0.0

**Comment by dbojdo** [comment]
@JindrichPilar by the way Doctrine DOES NOT support native JSON column type so far, probably it can be handle by providing custom `getSQLDeclaration` implementation but we must be careful about the platform itself, versions, etc. Please create another ticket if you'd like to handle this.

**Comment by dbojdo** [comment]
@JindrichPilar Actually, just double checked, it depends on the MySQL platform and Doctrine ORM version. Changed the SQL declaration, to use "JSON" from the Platform. Fixed in v2.0.1
